pkg/service/backup/service.go

```diff
@@ -1212,3 +1217,70 @@ func (s *Service) DeleteSnapshot(ctx context.Context, clusterID uuid.UUID, locat
 
 	return nil
 }
+
+// tableFilter is responsible for filtering tables that should be backed up.
+type tableFilter struct {
+	keyspaces  *ksfilter.Filter
+	dcs        *strset.Set
+	views      *strset.Set
+	liveNodes  *strset.Set
+	dcNodesCnt int
+	rd         scyllaclient.RingDescriber
+}
+
+func newTableFilter(f *ksfilter.Filter, dcs, views, liveDCNodes *strset.Set, dcNodesCnt int, rd scyllaclient.RingDescriber) tableFilter {
+	return tableFilter{
+		keyspaces:  f,
+		dcs:        dcs,
+		views:      views,
+		liveNodes:  liveDCNodes,
+		dcNodesCnt: dcNodesCnt,
+		rd:         rd,
+	}
+}
+
+// Returns true if table should be backed up.
+// Return error if table should, but can't be backed up.
+func (tf tableFilter) check(ctx context.Context, ks, tab string) (bool, error) {
```

Comment:
It's not intuitive to use this function in the code.
I don't know what it checks, must read the function body to understand and must read in what context it has been used.

The struct is called `tableFilter`, it could be set of Filter abstractions.
```
type filter interface {
   Filter(ctx context.Context, ks, table string) bool
}
```
}

Then you can have set of filters that implements this interface:
- keyspace
- view
- dataIntegrity (ring stuff) etc (although dataIntegrity is not a filter, it's a validity check)

The struct that is responsible for definitng the target (so the Target struct actually) could be created not just via:
```
	t := Target{}
```
but via sth like `BuildTarget(ctx context.Context, clusterID uuid.UUID, properties json.RawMessage) Target` or `NewTarget` in either backup/model.go ot backup/target.go. Worth to move GetTarget from service. It's not a simple getter, it has plenty of logic inside.

This `BuildTarget` (or whatever) can just be split per i.e.:
- build retention policy
- get the cluster data topology (don't know if it's accurate name, but just get all keyspaces and tables available on the cluster)
- filter according to the properties (which keyspaces, tables, no views)
- validate data integrity (check rings) against live nodes

WDYT ?

Reply:
Sure, I will try to make a general refactor of `GetTarget` tomorrow!